tests/test_URDFParser.py

```diff
@@ -0,0 +1,137 @@
+import unittest
+import os
+import xml.etree.ElementTree as ET
+import random
+
+import kinpy as kp
+import numpy as np
+from trip_kinematics import Utility, URDFParser, Robot
+import trip_kinematics
+
+
+def state_to_kp(state):
+    return {
+        "_".join(joint_name.split("_")[:-2]): value
+        for joint_name, value in state.items()
+    }
+
+
+def state_to_trip(state):
+    return state
+
+
+def initialize_state(robot):
+    return {
+        joint_name: 0
+        for joint_name in robot.get_actuated_state()
+    }
+
+
+def create_kinpy_chain(path):
+    with open(path, encoding='utf8') as file:
+        urdf_data_str = file.read()
+        return kp.build_chain_from_urdf(urdf_data_str)
+
+
+def get_joint_tree_dict(path):
+    tree = ET.parse(path)
+    root = tree.getroot()
+    joints = root.findall('joint')
+
+    # pylint: disable=protected-access
+    return URDFParser._build_joint_tree_dict(joints)
+
+
+def compare_urdf_trip_vs_kinpy(path, rng_states_count=10):
+    # Setup TriP robot using the URDF parser
+    try:
+        robot = Robot(URDFParser.from_urdf(path))
+        state_init = initialize_state(robot)
+    except ValueError as err:
+        raise ValueError(f'File {path} contains unsupported joint type ({err})') from err
+
+    # Setup kinpy chain
+    try:
+        chain_kinpy = create_kinpy_chain(path)
+    except KeyError as err:
+        raise ValueError(f'File {path} contains unsupported joint type ({err})') from err
+
+    test_states = [state_init]
+
+    for seed in range(rng_states_count):

```

Comment:
why uses a seed when using random?
This will always result in the same numbers.
Would seedless operation not be better?